### PR TITLE
WSAS-1981: JAX-WS/JAX-RS services does not get recognized when deployed into a virtual host

### PR DESCRIPTION
**Pull request by thusithathilina**

### CODE REVIEW

File: components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java [hunk 1]
```diff
@@ -177,11 +178,38 @@ public WebappMetadata getStoppedWebapp(String webappFileName, String hostName) {
...
+        }
+        webApplication.setServiceListPath(serviceListPathParam);
+    }
+
     private WebappMetadata getWebapp(WebApplication webApplication) {
         WebappMetadata webappMetadata;
         webappMetadata = new WebappMetadata();
 
         String appContext = WebAppUtils.checkJaxApplication(webApplication);
+        //Fix WSAS-1981
```
Comment (Niranjan-K) [inline components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java]: I don't think we need to add comment like this.

File: components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java [hunk 2]
```diff
@@ -177,11 +178,38 @@ public WebappMetadata getStoppedWebapp(String webappFileName, String hostName) {
...
+            Map<String, ? extends ServletRegistration> servletRegs =
+                    webApplication.getContext().getServletContext().getServletRegistrations();
+            for (ServletRegistration servletReg : servletRegs.values()) {
+                serviceListPathParam = servletReg.getInitParameter(serviceListPathParamName);
+                if (!"".equals(serviceListPathParam) || serviceListPathParam != null) {
+                    break;
+                }
+            }
+        }
+        if ("".equals(serviceListPathParam) || serviceListPathParam == null) {
```
Comment (Niranjan-K) [inline components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java]: Shouldn't be the null check be the first condition checked and then the empty string?

File: components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java [hunk 3]
```diff
@@ -177,11 +178,38 @@ public WebappMetadata getStoppedWebapp(String webappFileName, String hostName) {
         return webappMetadata;
     }
 
+    //TODO WSAS-2125
```
Comment (Niranjan-K) [inline components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java]: Better to have a small explanation along with the JIRA about the ToDo

Comment (supunmalinga) [inline components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java]: @thusithathilina, please note.

File: components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java [hunk 4]
```diff
@@ -177,11 +178,38 @@ public WebappMetadata getStoppedWebapp(String webappFileName, String hostName) {
...
+        }
+        webApplication.setServiceListPath(serviceListPathParam);
+    }
+
     private WebappMetadata getWebapp(WebApplication webApplication) {
         WebappMetadata webappMetadata;
         webappMetadata = new WebappMetadata();
 
         String appContext = WebAppUtils.checkJaxApplication(webApplication);
+        if(appContext != null){
```
Comment (callkalpa) [inline components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java]: Formatting issue

File: components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java [hunk 5]
```diff
@@ -177,11 +178,38 @@ public WebappMetadata getStoppedWebapp(String webappFileName, String hostName) {
...
+                serviceListPathParam = servletReg.getInitParameter(serviceListPathParamName);
+                if (serviceListPathParam != null || !"".equals(serviceListPathParam) ) {
+                    break;
+                }
+            }
+        }
+        if (serviceListPathParam == null || "".equals(serviceListPathParam)) {
+            serviceListPathParam = "/services";
+        } else {
+            serviceListPathParam = "";
```
Comment (sameera-jayasoma) [inline components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/WebappAdmin.java]: If the serviceListPathParam is not NULL, then serviceListPathParam always becomes empty.

**Pull request merged**